scylla/src/transport/retry_policy.rs

```
@@ -216,6 +216,172 @@ impl RetrySession for DefaultRetrySession {
+            // that these replicas are dead.
+            QueryError::DbError(
+                DbError::ReadTimeout {
+                    received,
+                    required,
+                    data_present,
+                    ..
+                },
+                _,
+            ) => {
+                if self.was_read_timeout_retry {
+                    RetryDecision::DontRetry
+                } else if received < required {
+                    self.was_read_timeout_retry = true;
+                    max_likely_to_work_cl(*received, cl)
+                } else if !*data_present {
+                    self.was_read_timeout_retry = true;
+                    RetryDecision::RetrySameNode(cl)
+                } else {
+                    RetryDecision::DontRetry
+                }
+            }
+            // Write timeout - coordinator didn't receive enough replies in time.
+            // Coordinator probably didn't detect the nodes as dead.
+            QueryError::DbError(
+                DbError::WriteTimeout {
+                    write_type,
+                    received,
+                    ..
+                },
+                _,
+            ) => {
+                if self.was_write_timeout_retry || !query_info.is_idempotent {
+                    RetryDecision::DontRetry
+                } else {
+                    self.was_write_timeout_retry = true;
+                    match write_type {
+                        WriteType::UnloggedBatch => {
+                            // Since only part of the batch could have been persisted,
+                            // retry with whatever consistency should allow to persist all
+                            max_likely_to_work_cl(*received, cl)
+                        }
+                        WriteType::BatchLog => RetryDecision::RetrySameNode(cl),
+
+                        _ => RetryDecision::DontRetry,
```

For simple writes I think we should ignore the error if at least one replica acknowledged the write.
That's what java driver does:

https://github.com/datastax/java-driver/blob/5234247cf2d60f8a1de49e490f7b208cfc019ed7/driver-core/src/main/java/com/datastax/driver/core/policies/DowngradingConsistencyRetryPolicy.java#L146-L163